File: Pokémon 3D Server/Pokémon 3D Server/Class/Setting/Data List/MuteList.vb
Public Class MuteList
    Public Property Name As String
    Public Property GameJoltID As Integer
    Public Property MuteReason As String
    Public Property MuteStartTime As Date
    Public Property MuteDuration As Integer

    Public ReadOnly Property BanRemainingTime As String
        Get
            Dim BanRemainingTimes As TimeSpan
            Dim BanRemainingText As String = Nothing

            If MuteDuration > 0 Then
                BanRemainingTimes = MuteStartTime.AddSeconds(MuteDuration) - Date.Now
                If BanRemainingTimes.Days = 1 Then
                    BanRemainingText &= BanRemainingTimes.Days & " day "
                ElseIf BanRemainingTimes.Days > 1
                    BanRemainingText &= BanRemainingTimes.Days & " days "
                End If

                If BanRemainingTimes.Hours = 1 AndAlso BanRemainingTimes.Days = 0 Then
                    BanRemainingText &= BanRemainingTimes.Hours & " hour "
                ElseIf BanRemainingTimes.Hours > 1 AndAlso BanRemainingTimes.Days = 0 Then
                    BanRemainingText &= BanRemainingTimes.Hours & " hours "
                End If

                If BanRemainingTimes.Minutes = 1 AndAlso BanRemainingTimes.Days = 0 Then
                    BanRemainingText &= BanRemainingTimes.Minutes & " minute "
                ElseIf BanRemainingTimes.Minutes > 1 AndAlso BanRemainingTimes.Days = 0 Then
                    BanRemainingText &= BanRemainingTimes.Minutes & " minutes "
                End If

                If BanRemainingTimes.Seconds = 1 AndAlso BanRemainingTimes.Days = 0 Then
                    BanRemainingText &= BanRemainingTimes.Seconds & " second "
                ElseIf BanRemainingTimes.Seconds > 1 AndAlso BanRemainingTimes.Days = 0 Then
                    BanRemainingText &= BanRemainingTimes.Seconds & " seconds "
                End If
            Else
                BanRemainingText = "Permanent"
            End If
            Return BanRemainingText
        End Get
    End Property

    Public Sub New(ByVal Name As String, ByVal GameJoltID As Integer, ByVal MuteReason As String, ByVal MuteStartTime As Date, ByVal MuteDuration As Integer)
        Me.Name = Name
        Me.GameJoltID = GameJoltID
        Me.MuteReason = MuteReason
        Me.MuteStartTime = MuteStartTime
        Me.MuteDuration = MuteDuration
    End Sub
End Class

File: Pokemon.3D.Server.Core/Server_Client_Listener/Settings/Data/MuteList.cs
using System;

namespace Pokemon_3D_Server_Core.Server_Client_Listener.Settings.Data
{
    /// <summary>
    /// Class containing MuteList data
    /// </summary>
    public class MuteList : CommonData
    {
        /// <summary>
        /// Get/Set MuteList StartTime
        /// </summary>
        public DateTime StartTime { get; set; }

        /// <summary>
        /// Get/Set MuteList Duration
        /// </summary>
        public int Duration { get; set; }

        /// <summary>
        /// Get MuteList Remaining Time
        /// </summary>
        public string RemainingTime
        {
            get
            {
                string ReturnText = null;

                if (Duration > 0)
                {
                    if (StartTime.AddSeconds(Duration) > DateTime.Now)
                    {
                        TimeSpan RemainingTime = StartTime.AddSeconds(Duration) - DateTime.Now;

                        if (RemainingTime.Days > 1)
                        {
                            ReturnText = RemainingTime.Days.ToString() + " Days";
                        }
                        else if (RemainingTime.Days == 1)
                        {
                            ReturnText = "1 Day";
                        }
                        else
                        {
                            if (RemainingTime.Hours == 1)
                            {
                                ReturnText += "1 Hour ";
                            }
                            else if (RemainingTime.Hours > 1)
                            {
                                ReturnText += RemainingTime.Hours.ToString() + " Hour ";
                            }

                            if (RemainingTime.Minutes == 1)
                            {
                                ReturnText += "1 Minute ";
                            }
                            else if (RemainingTime.Minutes > 1)
                            {
                                ReturnText += RemainingTime.Minutes.ToString() + " Minutes ";
                            }
                            else if (RemainingTime.Minutes == 0 && RemainingTime.TotalSeconds > 60)
                            {
                                ReturnText += "0 Minute ";
                            }

                            if (RemainingTime.Seconds == 1)
                            {
                                ReturnText += "1 Second";
                            }
                            else if (RemainingTime.Seconds > 1)
                            {
                                ReturnText += RemainingTime.Seconds.ToString() + " Seconds";
                            }
                            else if (RemainingTime.Seconds == 0)
                            {
                                ReturnText += "0 Second";
                            }
                        }
                    }
                    else
                    {
                        ReturnText = null;
                    }
                }
                else
                {
                    ReturnText = "Permanent";
                }
                return ReturnText;
            }
        }

        /// <summary>
        /// New MuteList
        /// </summary>
        /// <param name="Name">Player Name</param>
        /// <param name="GameJoltID">Player GameJolt ID</param>
        /// <param name="Reason">MuteList Reason</param>
        /// <param name="StartTime">MuteList StartTime</param>
        /// <param name="Duration">MuteList Duration</param>
        public MuteList(string Name, int GameJoltID, string Reason, DateTime StartTime, int Duration)
        {
            this.Name = Name;
            this.GameJoltID = GameJoltID;
            this.Reason = Reason;
            this.StartTime = StartTime;
            this.Duration = Duration;
        }
    }
}